cmake/test-targets.cmake

```
@@ -93,10 +93,10 @@ endif ()
 find_program(PROVE_PATH prove)
 find_program(UNBUFFER_PATH unbuffer)
 if (PROVE_PATH)
-    set(INVOKE_TEST_ARGS --prove-tool "${PROVE_PATH}" --make-tool "${CMAKE_MAKE_PROGRAM}" --unbuffer-tool "${UNBUFFER_PATH}" --build-directory "${CMAKE_CURRENT_BINARY_DIR}" t)
+    set(INVOKE_TEST_ARGS --prove-tool "${PROVE_PATH}" --make-tool "${CMAKE_MAKE_PROGRAM}" --unbuffer-tool "${UNBUFFER_PATH}" --build-directory "${CMAKE_CURRENT_BINARY_DIR}")
     add_test(
         NAME test-perl-testsuite
-        COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/tools/invoke-tests" ${INVOKE_TEST_ARGS}
+        COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/tools/invoke-tests" ${INVOKE_TEST_ARGS} TESTS=t
```

Seems like this trips up the "test" target somehow. Not sure yet how to fix it. Or maybe I need to adapt `invoke-tests` afterall 🤔

---

Well, invoke-tests takes the test file paths as command line arguments directly *but* there is also an *environment* variable TESTS so you would need to use

```suggestion
        COMMAND "TESTS=t ${CMAKE_CURRENT_SOURCE_DIR}/tools/invoke-tests" ${INVOKE_TEST_ARGS}
```

if cmake allows to use a variable assignment for an application. Otherwise use

```suggestion
        COMMAND "env TESTS=t ${CMAKE_CURRENT_SOURCE_DIR}/tools/invoke-tests" ${INVOKE_TEST_ARGS}
```

and that would work for invoke-tests as the command line parameter if specified takes precedence over an environment variable.
However then you again would harcode the variable TESTS and do not have a possibilty to override that on all so you would not have gained anything. How about just changing the README to recommend `prove -I. t/15-logging.t t/28-signalblocker.t` for the customized command?